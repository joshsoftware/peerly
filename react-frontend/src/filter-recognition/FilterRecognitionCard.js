import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import { Card } from "core-components/grid/GridComponent";
import { Form } from "core-components/form/FormComponent";
import { Button } from "core-components/button/ButtonComponent";
import AsyncSelect from "core-components/autocomplete/AutoCompleteComponent";

const CardBody = styled(Card.Body)`
  width: auto;
  padding: 0%;
`;

const Header = styled.div`
  width: auto;
  height: 20%;
  background: #87bcbf;
  border-radius: 50px 50px 0px 0px;
  text-align: center;
`;

const FilterCard = styled(Card)`
  border-radius: 50px;
`;

const FilterRecognitionCard = ({
  coreValueList,
  givenForList,
  givenByList,
  promiseCoreValueOptions,
  promiseGivenByOptions,
  promiseGivenForOptions,
  onSubmit,
}) => (
  <FilterCard className="h-100">
    <CardBody>
      <Header className="text -center h4">Filter</Header>
      <Form className="p-3" onSubmit={onSubmit} data-testid="filterform">
        <Form.Group controlId="filterCoreValue">
          <Form.Label>Core Value</Form.Label>
          <AsyncSelect
            defaultOptions={coreValueList}
            loadOptions={promiseCoreValueOptions}
            name="core_value"
            placeholder="select Core Value"
          />
        </Form.Group>
        <Form.Group controlId="filterGivenFor">
          <Form.Label>Given For</Form.Label>
          <AsyncSelect
            defaultOptions={givenForList}
            loadOptions={promiseGivenForOptions}
            name="given_for"
            placeholder="select given for"
          />
        </Form.Group>

        <Form.Group controlId="filterGivenBy">
          <Form.Label>Given By</Form.Label>
          <AsyncSelect
            defaultOptions={givenByList}
            loadOptions={promiseGivenByOptions}
            name="given_by"
            placeholder="select given by"
          />
        </Form.Group>
        <div className="text-center">
          <Button type="submit" variant="success">
            Apply Filter
          </Button>
        </div>
      </Form>
    </CardBody>
  </FilterCard>
);

FilterRecognitionCard.propTypes = {
  coreValueList: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.any.isRequired,
      value: PropTypes.shape({
        id: PropTypes.number.isRequired,
        text: PropTypes.string,
      }),
    })
  ),
  givenForList: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.any.isRequired,
      value: PropTypes.shape({
        id: PropTypes.number.isRequired,
        text: PropTypes.string,
      }),
    })
  ),
  givenByList: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.any.isRequired,
      value: PropTypes.shape({
        id: PropTypes.number.isRequired,
        text: PropTypes.string.isRequired,
      }),
    })
  ),
  promiseCoreValueOptions: PropTypes.func.isRequired,
  promiseGivenForOptions: PropTypes.func.isRequired,
  promiseGivenByOptions: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default FilterRecognitionCard;
