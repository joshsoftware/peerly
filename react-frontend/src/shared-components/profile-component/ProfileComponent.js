import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Image from "core-components/image/ImageComponent";
import { Form } from "core-components/form/FormComponent";
import { Col, Row } from "core-components/grid/GridComponent";

const ImageComponent = styled(Image)`
  height: inherit;
  width: inherit;
}
`;

const ProfileComponent = ({ className, text, imageSrc, key, alt }) => (
  <Row className={className} key={key}>
    <Col className="h-100 w-50 d-flex justify-content-center align-items-center">
      <ImageComponent
        src={imageSrc}
        alt={alt}
        fluid={false}
        roundedCircle={true}
      />
    </Col>
    <Col className="w-50 text-center">
      <Form.Label>{text}</Form.Label>
    </Col>
  </Row>
);

ProfileComponent.propTypes = {
  className: PropTypes.string,
  imageSrc: PropTypes.string.isRequired,
  text: PropTypes.string,
  name: PropTypes.string,
  key: PropTypes.number,
  alt: PropTypes.string,
};

ProfileComponent.defaultProps = {
  imageSrc: require("./profile.png"),
  text: "Employee Name",
  className: "d-flex justify-content-center align-items-center h-100",
};

export default ProfileComponent;
