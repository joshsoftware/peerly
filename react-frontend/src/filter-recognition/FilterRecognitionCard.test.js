import React from "react";
import { render } from "@testing-library/react";

import FilterRecognitionCard from "filter-recognition/FilterRecognitionCard";
import ProfileComponent from "shared-components/profile-component/ProfileComponent.js";

describe("FilterRecognitionCard component test", () => {
  const list = [
    {
      label: <ProfileComponent text="Jitendra Bunde" />,
      value: {
        id: 1,
        text: "Jitendra Bunde",
      },
    },
  ];

  const promiseOptionfunction = () => {
    return new Promise((resolve) => {
      setTimeout(() => {
        resolve(list);
      }, 1000);
    });
  };

  const onSubmit = () => {
    return true;
  };

  test("should render FilterRecognitionCard component", () => {
    const { asFragment } = render(
      <FilterRecognitionCard
        coreValueList={list}
        promiseCoreValueOptions={promiseOptionfunction}
        givenForList={list}
        promiseGivenForOptions={promiseOptionfunction}
        givenByList={list}
        promiseGivenByOptions={promiseOptionfunction}
        onSubmit={onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
