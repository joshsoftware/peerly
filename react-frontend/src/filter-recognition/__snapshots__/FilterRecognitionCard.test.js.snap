// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRecognitionCard component test should render FilterRecognitionCard component 1`] = `
<DocumentFragment>
  .c1 {
  width: auto;
  padding: 0%;
}

.c2 {
  width: auto;
  height: 20%;
  background: #87bcbf;
  border-radius: 50px 50px 0px 0px;
}

.c0 {
  border-radius: 50px;
}

<div
    class="c0 h-100 card"
  >
    <div
      class="c1 card-body"
    >
      <div
        class="c2 d-flex align-content-center justify-content-center flex-wrap h3"
      >
        Filter
      </div>
      <form
        class="p-3"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="filterCoreValue"
          >
            Core Value
          </label>
          <div
            class=" css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                >
                  select Core Value
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="core_value"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="filterGivenFor"
          >
            Given For
          </label>
          <div
            class=" css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                >
                  select given for
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-3-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="given_for"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="filterGivenBy"
          >
            Given By
          </label>
          <div
            class=" css-2b097c-container"
          >
            <div
              class=" css-yk16xz-control"
            >
              <div
                class=" css-g1d714-ValueContainer"
              >
                <div
                  class=" css-1wa3eu0-placeholder"
                >
                  select given by
                </div>
                <div
                  class="css-b8ldur-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-4-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="given_by"
              type="hidden"
              value=""
            />
          </div>
        </div>
        <div
          class="text-center"
        >
          <button
            class="btn btn-success"
            type="submit"
          >
            Apply Filter
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
